graf2d/asimage/src/TASImage.cxx

```diff
@@ -6308,7 +6308,7 @@ void TASImage::SetTitle(const char *title)
    int stop = fTitle.Index("*/") - 1;
 
    if ((start > 0) && (stop - start > 0)) {
-      fTitle.Replace(start, stop - start, title);
+      fTitle.Replace(start, stop - start, title.data());
```

Comment:
`std::string_view::data` is not 'null' terminated at the end of the 'view' and will return all the data from the string being view (starting from where the view starts):
```
root [0] std::string longname("this is a long name");
root [1] std::string_view v(longname.c_str()+5, 4);
root [2] v
(std::string_view &) "is a"[4]
root [3] v.data()
(std::basic_string_view<char, std::char_traits<char> >::const_pointer) "is a long name"
```

So any places inside the function where the `const char*` is being used needs to be reviewed to see:

1. if there is an overload of the function being called that takes a `std::string_view`
2. else consider adding that overload and/or change the signature to take a `std::string_view` (and recursively apply this source code change algorithm).
3. else (avoid if possible as it is much less performant) copy the `std::string_view` into a `std::string` before passing it.

Reply:
This is the problem with `std::string_view` and the reason why we didn't adopt it much yet. It's quite some work to avoid accidental copies, not sure how to argue it's worth it! Especially for our Python users there is no added benefit because any string-type class is pythonized to implicitly convert to `const char*`.

I was thinking about this years ago and realized there was not good backwards compatible solution without added boilerplate code, other than adding a new adapter class. This was done for RooFit:
https://github.com/root-project/root/blob/master/roofit/roofitcore/inc/RooStringView.h

Maybe this `RooStringView` can be generalized for the use in the rest of ROOT, like in TNamed? And maybe it should not have "StringView" in the name because it's not necessarily a view but more a `ROOT::StringAdapter`? But then these non-standard classes confuse the users..

Of course that doesn't help with `virtual` functions. We should keep the original virtual overloads there.

So that's why personally I concluded it was better to not change the status quo.

Reply:
I guess option 1 can be easily done for the "Replace" function. For all the other changes in this PR at this end the std::string_view is passed to a TString at the end and as such is copied.